src/data/zsqrtd/basic.lean

```diff
@@ -571,4 +575,40 @@ instance : linear_ordered_semiring ℤ√d := by apply_instance
 instance : ordered_semiring ℤ√d        := by apply_instance
 
 end
+
+lemma norm_eq_zero {d : ℤ} (h : 0 ≤ d) (h_nonsquare : ∀ n : ℤ, d ≠ n*n) (a : ℤ√d) :
```

Comment:
This should work for negative `d` as well. The nonsquare assumption is sufficient. (Compare with complex numbers, where you take `d = -1`.)

Reply:
Indeed it should, but the lemmas I'm using only work for \N, and I don't want to refactor this whole file - plus the non-negative case is the only one relevant to `to_real`.

Do you want me to leave a TODO comment in lieu of the refactor?

Reply:
```suggestion
/-- TODO: This lemma only assumes `0 ≤ d` because it is needed by `zsqrtd.divides_sq_eq_zero_z`.
Both lemmas should be adjusted to not require this. -/
lemma norm_eq_zero {d : ℤ} (h : 0 ≤ d) (h_nonsquare : ∀ n : ℤ, d ≠ n*n) (a : ℤ√d) :
```

Reply:
Thanks for generalizing that. I guess `zsqrtd.divides_sq_eq_zero_z` can be generalized in another PR, in the unlikely event that anyone cares!